bindings/python/setup.py

```diff
@@ -250,7 +256,10 @@ def build_extension_with_b2(self):
             log.info(config.read())
             config.close()
             args.append(f"--project-config={config.name}")
-            self.spawn(["b2"] + args)
+
+            python_binding_dir = pathlib.Path(__file__).parent.absolute()
+            log.info(" ".join(["b2"] + args))
+            subprocess.run(["b2"] + args, cwd=python_binding_dir, check=True)
```

Comment:
Note that there's a lot of logic in `spawn()`: https://github.com/pypa/distutils/blob/35e1c9d4d788094c254485dfdefa0c24cd5820ee/distutils/spawn.py#L23

I'm not sure what it all does.

To keep it intact, I think it's also reasonable to do

```python
cwd = os.getcwd()
try:
    os.chdir(python_binding_dir)
    self.spawn(["b2"] + args)
finally:
    os.chdir(cwd)
```

I don't see a builtin `pushd`-like contextmanager.

Reply:
I don't think anything `spawn()` does is important. Most of it is logging and checking environment variables for MacOS.
I looked around for `pushd` or directory stack solutions for python, but I think it's a lot more reliable to leave cwd untouched, as my current patch does. It's a global variable that anything can depend on, so I think it's best to leave it alone.